src/jit/inline.cpp

```diff
@@ -0,0 +1,93 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "jitpch.h"
+#ifdef _MSC_VER
+#pragma hdrstop
+#endif
+
+static const char* InlineExplanations[] = 
+{
+    "unused initial",
+#define INLINE_OBSERVATION(name, type, description, impact, target) description,
+#include "inline.def"
+#undef INLINE_OBSERVATION
+    "unused final"
+};
+
+static const InlineTarget InlineTargets[] =
+{
+    InlineTarget::CALLSITE,
+#define INLINE_OBSERVATION(name, type, description, impact, target) InlineTarget::target,
+#include "inline.def"
+#undef INLINE_OBSERVATION
+    InlineTarget::CALLSITE
+};
+
+static const InlineImpact InlineImpacts[] =
+{
+    InlineImpact::FATAL,
+#define INLINE_OBSERVATION(name, type, description, impact, target) InlineImpact::impact,
+#include "inline.def"
+#undef INLINE_OBSERVATION
+    InlineImpact::FATAL,
+};
+
+const char* GetDescriptionString(InlineObservation obs)
+{
+    assert((obs > InlineObservation::UNUSED_INITIAL) &&
+           (obs < InlineObservation::UNUSED_FINAL));
+    return InlineExplanations[static_cast<int>(obs)];
+}
+
+InlineTarget GetTarget(InlineObservation obs)
+{
+    assert((obs > InlineObservation::UNUSED_INITIAL) &&
+           (obs < InlineObservation::UNUSED_FINAL));
+    return InlineTargets[static_cast<int>(obs)];
+}
+
+const char* GetTargetstring(InlineObservation obs)
+{
+    InlineTarget t = GetTarget(obs);
+    switch (t) 
+    {
+        case InlineTarget::CALLER:
+            return "caller";
+        case InlineTarget::CALLEE:
+            return "callee";
+        case InlineTarget::CALLSITE:
+            return "call site";
+        default:
+            return "unexpected target";
+    }
+}
+
+InlineImpact GetImpact(InlineObservation obs)
+{
+    assert((obs > InlineObservation::UNUSED_INITIAL) &&
+           (obs < InlineObservation::UNUSED_FINAL));
+    return InlineImpacts[static_cast<int>(obs)];
+}
+
+const char* GetImpactString(InlineObservation obs)
+{
+    InlineImpact i = GetImpact(obs);
+    switch (i) 
+    {
+        case InlineImpact::FATAL:
+            return "correctness -- fatal";
+        case InlineImpact::SERIOUS:
+            return "correctness -- serious";
+        case InlineImpact::LIMITATION:
+            return "correctness -- jit limitation";
+        case InlineImpact::PERFORMANCE:
+            return "performance";
+        case InlineImpact::INFORMATION:
+            return "information";
+        default:
+            return "unexpected impact";
+    }
+}
+
```

Comment:
Not clear yet. Currently they're not used at all. I'll keep them as is and when I get around to using them I'll make them debug if possible.